# R packages and sources


## look source code of R packages

- [lookup](https://github.com/jimhester/lookup), **package**, Lookup R full  
  function definitions, including compiled code, S3 and S4 methods.
- [Accessing R Source](https://github.com/jennybc/access-r-source), source of R
  function, S3 class and compiled code
  
## visualization

- [ggpointdensity](https://github.com/LKremer/ggpointdensity), **package**, 
  `geom_pointdensity()`: A cross between a scatter plot and a 2D density plot. 
  If you have lots of data points on top of each other, `geom_point()` fails to 
  give you an estimate of how many points are overlapping. `geom_density2d()` and
  geom_bin2d() solve this issue, but they make it impossible to investigate 
  individual outlier points, which may be of interest.
  
## data science

## Bioinformatics

### genomic

- [GenVisR](https://github.com/griffithlab/GenVisR/issues), **package**,
  visulization for genomic data

## microbe/metagenomic

### R package

- [phyloseq](https://github.com/joey711/phyloseq), phyloseq: An R 
  Package for Reproducible Interactive Analysis and Graphics of Microbiome Census
  Data
- [microbiome](https://github.com/microbiome/microbiome), tools for
  microbiome analysis
- [MicrobiomeR](https://github.com/vallenderlab/MicrobiomeR), An R package for
  microbiome analysis that incorporates phyloseq, metacoder, taxa, and microbiome 
  in order to standardize and simplify common microbiome workflows
- [curatedMetagemicData](https://github.com/waldronlab/curatedMetagenomicData), 
  Curated and processed metagenomic data through ExperimentHub 
- [metacoder](https://github.com/grunwaldlab/metacoder), Parsing, Manipulation,
  and Visualization of Metabarcoding/Taxonomic data
- [microbiomeSeq](https://github.com/umerijaz/microbiomeSeq),  enhance the 
  available statistical analysis procedures in R by providing more analysis
  produre and visualisation of results for microbial communities data obtained
  from 16S rRNA
- [metavizR](https://github.com/epiviz/metavizr),  implements two-way
  communication between the R/Bioconductor environment and the metaviz web app
  for interactive visualization of microbiome sequencing result
- [metagenomeFeatures](https://github.com/HCBravoLab/metagenomeFeatures),
  exploring the taxonomic annotations for a marker-gene metagenomic sequence
  dataset. The package can be used to explore the taxonomic composition of a
  marker-gene database or annotated sequences from a marker-gene metagenome
  experiment
- [ShotgunFunctionalizedR](http://shotgun.math.chalmers.se/),  tools for
  importing, annotating and visualising metagenomic data produced by shotgun
  high-throughput sequencing. ShotgunFunctionalizeR contains several statistical
  procedures for assessing functional differences between samples, both for
  individual genes and for entire pathways. In addition to standard and
  previously published methods, we have developed and implemented a novel
  approach based on a Poisson model
- [melonnpan](https://github.com/biobakery/melonnpan), Model-based Genomically
  Informed High-dimensional Predictor of Microbial Community Metabolic Profiles
- [pavian](https://github.com/fbreitwieser/pavian), a interactive browser
  application for analyzing and visualization metagenomics classification results
  from classifiers such as Kraken, Centrifuge and MetaPhlAn. Pavian also provides
  an alignment viewer for validation of matches to a particular genome.
- [microbiomeViz](https://github.com/lch14forever/microbiomeViz), Visualize
  microbiome data with black magic ggtree, lefse visualization
- [PathoStat](https://github.com/mani2012/PathoStat), Statistical Microbiome
  Analysis on metagenomics results from sequencing data samples
- [SpiecEasi](https://github.com/zdk123/SpiecEasi), Sparse InversE Covariance
  estimation for Ecological Association and Statistical Inference
- [microbiomeutilities](https://github.com/microsud/microbiomeutilities), a
  wrapper tool using phyloseq and microbiome R packages. Apart for some simple
  scripts, this package has a single function generates a HTML report with
  preliminary QC, Alpha Diversity, Ordination and Composition analysis of OTU
  tables
- [Tax4Fun](http://tax4fun.gobics.de/), predicting functional profiles from
  metagenomic 16S rRNA data
- [taxize](https://github.com/ropensci/taxize), search over many taxonomic data
  sources for species names (scientific and common) and download up and
  downstream taxonomic hierarchical information - among other things
- [taxa](https://github.com/ropensci/taxa), taxonomic classes and functions to manipulate them

### code

- [microbiome_helper](https://github.com/LangilleLab/microbiome_helper),some R
  scripts to help process and automate various microbiome and metagenomic
  bioinformatic tools, e.g. data2 output to R


## Awesome list

- [awesome-R](https://github.com/qinwf/awesome-R): A curated list of awesome R 
  packages and tools
- [DataScienceR](https://github.com/ujjwalkarn/DataScienceR): a curated list of R
  tutorials for Data Science, NLP and Machine Learning

## ebook

Repository [R-ebbok](https://github.com/yiluheihei/R-ebook) stored all the r programming e-books I collected.


## Contributing

Your contributions are always welcome!
